x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -1090,6 +1090,37 @@ public void testMatchCommand() throws Exception {
         // TODO Keep adding tests for all unsupported commands
     }
 
+    public void testQueryStringFunctionsNotAllowed() throws Exception {
+        assumeTrue("skipping because QSTR is not enabled", EsqlCapabilities.Cap.QSTR_FUNCTION.isEnabled());
+        assertEquals("1:24: Full text functions cannot be used after LIMIT", error("from test | limit 10 | where qstr( \"Anna\")"));
+        assertEquals("1:13: Full text functions cannot be used after SHOW", error("show info | where qstr( \"8.16.0\")"));
+        assertEquals("1:17: Full text functions cannot be used after ROW", error("row a= \"Anna\" | where qstr( \"Anna\")"));
+        assertEquals("1:26: Full text functions cannot be used after EVAL", error("from test | eval z = 2 | where qstr( \"Anna\")"));
+        assertEquals(
+            "1:43: Full text functions cannot be used after DISSECT",
+            error("from test | dissect first_name \"%{foo}\" | where qstr( \"Connection\")")
+        );
+        assertEquals("1:27: Full text functions cannot be used after DROP", error("from test | drop emp_no | where qstr( \"Anna\")"));
+        assertEquals(
+            "1:35: Full text functions cannot be used after EVAL",
+            error("from test | eval n = emp_no * 3 | where qstr( \"Anna\")")
+        );
+        assertEquals(
+            "1:44: Full text functions cannot be used after GROK",
+            error("from test | grok last_name \"%{WORD:foo}\" | where qstr( \"Anna\")")
+        );
+        assertEquals("1:27: Full text functions cannot be used after KEEP", error("from test | keep emp_no | where qstr( \"Anna\")"));
+    }
+
+    public void testQueryStringFunctionOperands() throws Exception {
+        assumeTrue("skipping because QSTR is not enabled", EsqlCapabilities.Cap.QSTR_FUNCTION.isEnabled());
+        assertEquals("1:19: argument of [qstr(10)] must be [string], found value [10] type [integer]", error("from test | where qstr(10)"));
+        assertEquals(
+            "1:19: Query in QSTR function needs to be statically resolved. References to fields are not allowed.",
+            error("from test | where qstr(first_name)")
+        );
+    }
+
```

Comment:
Add some negative testcases here for:

1. Cover more invalid input data types - all of the available DataTypes are defined in `DataType`. `AbstractFunctionTestCases` has the infrastructure to cover invalid input data types, it would be nice to have it, without it, `VerifierTests` is a good place to add them.
2. null as input
3. Invalid string literals as input